compiler/rustc_typeck/src/collect.rs

```diff
@@ -1711,10 +1713,21 @@ pub fn const_evaluatable_predicates_of<'tcx>(
         collector.curr_span = span;
         pred.visit_with(&mut collector);
     }
-    warn!("const_evaluatable_predicates_of({:?}) = {:?}", def_id, collector.ct);
+
+    match tcx.def_kind(def_id) {
+        DefKind::Fn | DefKind::AssocFn => {
```

Comment:
In the case of `AssocFn` we should probably also visit the signature of the `impl` (there may be constants in the optional trait or in the type)

Reply:
visiting the self ty and trait of impls now, this adds the bound to the impl itself instead of the function, which is probably what we want here